autils/data_structures.py

```
@@ -0,0 +1,573 @@
+
+        Executes all registered functions with their associated arguments.
+        If any function raises an exception, it is logged and execution
+        continues with the remaining functions. Functions are called in
+        LIFO order (last registered, first executed).
+        """
+        while self._items:
+            item = self._items.pop()
+            try:
+                func, args, kwargs = item
+                func(*args, **kwargs)
+            except:  # Ignore all exceptions pylint: disable=W0702
```

Ruff (0.12.2)

Do not use bare `except`

(E722)